### PR TITLE
[chip-tool] Move WebSocketServer code to src/app/tests/suites/server …

### PR DESCRIPTION
**Pull request by vivien-apple**
…so it can be reused

#### Problem

This PR just moves `chip-tool` web socket server code to a place where other apps can used it.

### CODE REVIEW

**Comment by vivien-apple** [comment]
> src/app is taken as part of Matter core SDK in coverage statistic, if WebSocketServer is only reused by example apps, why not move to some place under examples, such as examples/common?

`src/app/tests/suites/` has since a long time be a place where code related to the test harness lives. If that is really an issue I believe I can moved it to `examples/common`...

**Comment by yufengwangca** [comment]
> > src/app is taken as part of Matter core SDK in coverage statistic, if WebSocketServer is only reused by example apps, why not move to some place under examples, such as examples/common?
> 
> `src/app/tests/suites/` has since a long time be a place where code related to the test harness lives. If that is really an issue I believe I can moved it to `examples/common`...

Currently, WebSocketServer is under 'examples/chip-tool/commands/interactive', so I think this is not used only for the testing.

The idea is we want to set a clear boundary between the example code and core SDK code, we conducts daily coverage statistic for Matter Core SDK for our internal reference, we don't count the code coverage in the examples now, so it is better to keep all example code outside of core SDK directories.

File: examples/chip-tool/BUILD.gn [hunk 1]
```diff
@@ -87,13 +87,11 @@ static_library("chip-tool-utils") {
...
 
   if (config_use_interactive_mode) {
-    sources += [
-      "commands/interactive/InteractiveCommands.cpp",
-      "commands/interactive/WebSocketServer.cpp",
-      "commands/interactive/WebSocketServer.h",
-      "commands/interactive/WebSocketServerDelegate.h",
+    sources += [ "commands/interactive/InteractiveCommands.cpp" ]
+    deps += [
+      "${chip_root}/src/app/tests/suites/server:test_server",
```
Comment (andy31415) [inline examples/chip-tool/BUILD.gn:92]: What do you think of naming targets at least being `${chip_root}/src/app/tests/suites/server:websocket` to be explicit on what this includes?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chip-tool/BUILD.gn (hunk 1 above)
Comment (vivien-apple) [reply examples/chip-tool/BUILD.gn]: I moved/renamed the component as : `examples/common/websocket-server`

**Comment by vivien-apple** [comment]
> > > src/app is taken as part of Matter core SDK in coverage statistic, if WebSocketServer is only reused by example apps, why not move to some place under examples, such as examples/common?
> > 
> > 
> > `src/app/tests/suites/` has since a long time be a place where code related to the test harness lives. If that is really an issue I believe I can moved it to `examples/common`...
> 
> Currently, WebSocketServer is under 'examples/chip-tool/commands/interactive', so I think this is not used only for the testing.
> 
> The idea is we want to set a clear boundary between the example code and core SDK code, we conducts daily coverage statistic for Matter Core SDK for our internal reference, we don't count the code coverage in the examples now, so it is better to keep all example code outside of core SDK directories.

I have moved the component in `examples/common`...

**Review by andy31415 — APPROVED** [review]

**Pull request merged**